### PR TITLE
[SDL-0205] Avoid Custom button subscription in case HMI incompatibility

### PR DESCRIPTION
**Pull request by VjKlepikov**
**Note:** Scripts are based on implementation for [SDL-0192 Button Subscription response from HMI](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0192-button_subscription_response_from_hmi.md) feature. 
So this PR should be merged AFTER [#2564](https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2564)

ATF Test Scripts to check [#2789](https://github.com/smartdevicelink/sdl_core/issues/2789)

This PR is **[ready]** for review.

### Summary
Scripts for [SDL-0205 Avoid Custom button subscription in case HMI incompatibility](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0205-Avoid_custom_button_subscription_when_HMI_does_not_support.md) feature

### ATF version
Integration ATF branch: [integrated/sdl_0192_button_subscription_response_from_hmi](https://github.com/smartdevicelink/sdl_atf/tree/integrated/sdl_0192_button_subscription_response_from_hmi) which contains:
- [pull/232](https://github.com/smartdevicelink/sdl_atf/pull/232) Bump default app version for 8.0 release
- [pull/233](https://github.com/smartdevicelink/sdl_atf/pull/233) [SDL-0192] Button subscription response from hmi

### Changelog
- Added Test Scripts and Test Set for feature

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Comment by ashwink11** [comment]
@jordynmackool this PR is ready for review and is approved by Ford.

File: test_scripts/API/ButtonSubscription/commonButtonSubscription.lua [hunk 1]
```diff
@@ -0,0 +1,561 @@
...
+--[[ @addButtonToCapabilities: add button to HMI capabilities
+--! @parameters:
+--! pButtonCapabilities - button capabilities
+--! @return: returns HMI Table with supported button
+--]]
+function m.addButtonToCapabilities(pButtonCapabilities)
+  local hmiValues = m.getDefaultHMITable()
+  for i, buttonNameTab in pairs(hmiValues.Buttons.GetCapabilities.params.capabilities) do
+    if (buttonNameTab.name == pButtonCapabilities.name) then
+      table.remove(hmiValues.Buttons.GetCapabilities.params.capabilities, i)
```
[the remark this answers is not in the document]
Comment (dboltovskyi) [reply test_scripts/API/ButtonSubscription/commonButtonSubscription.lua:443]: @iCollin Please find update in 0d992b5

File: test_scripts/API/ButtonSubscription/commonButtonSubscription.lua [hunk 2]
```diff
@@ -0,0 +1,561 @@
...
+--[[ @removeButtonFromHMICapabilitiesFile: remove button support from hmi_capabilities.json file
+--! @parameters:
+--! pButtonName - button name
+--! @return: none
+--]]
+function m.removeButtonFromHMICapabilitiesFile(pButtonName)
+  local hmiCapTbl = m.getHMICapabilitiesFromFile()
+  for i, buttonNameTab in pairs(hmiCapTbl.Buttons.capabilities) do
+    if (buttonNameTab.name == pButtonName) then
+      table.remove(hmiCapTbl.Buttons.capabilities, i)
```
[the remark this answers is not in the document]
Comment (dboltovskyi) [reply test_scripts/API/ButtonSubscription/commonButtonSubscription.lua:460]: @iCollin Please find update in 0d992b5

File: test_scripts/API/ButtonSubscription/commonButtonSubscription.lua [hunk 3]
```diff
@@ -0,0 +1,561 @@
...
+--[[ @addButtonToHMICapabilitiesFile: add button support to hmi_capabilities.json file
+--! @parameters:
+--! pButtonCapabilities - button capabilities
+--! @return: none
+--]]
+function m.addButtonToHMICapabilitiesFile(pButtonCapabilities)
+  local hmiCapTbl = m.getHMICapabilitiesFromFile()
+  for i, buttonNameTab in pairs(hmiCapTbl.Buttons.capabilities) do
+    if (buttonNameTab.name == pButtonCapabilities.name) then
+      table.remove(hmiCapTbl.Buttons.capabilities, i)
```
[the remark this answers is not in the document]
Comment (dboltovskyi) [reply test_scripts/API/ButtonSubscription/commonButtonSubscription.lua:476]: @iCollin Please find update in 0d992b5

**Comment by iCollin** [comment]
Could you please merge in develop?

**Comment by dboltovskyi** [comment]
@iCollin Please find update in 943d9e8

**Review by iCollin — APPROVED** [review]

**Pull request merged**